test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -171,7 +186,40 @@ func (r *DockerMachinePoolReconciler) SetupWithManager(ctx context.Context, mgr
 }
 
 func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) (ctrl.Result, error) {
-	pool, err := docker.NewNodePool(ctx, r.Client, cluster, machinePool, dockerMachinePool)
+	log := ctrl.LoggerFrom(ctx)
+
+	dockerMachineList, err := getDockerMachines(ctx, r.Client, *dockerMachinePool)
```

Comment:
Looks like we (almost) have the same code here and in reconcileNormal. Is the difference intentional? Would it make sense to move this into a helper func to de-duplicate?

Reply:
Yeah, I think we can try to move this to a helper.